src/Tasks/GenerateResource.cs

```diff
@@ -2992,40 +3028,57 @@ private void ReadResources(String filename, bool shouldUseSourcePath, String out
+                            }
+                            else
+                            {
+                                resXReader = new ResXResourceReader(filename);
+                            }
 
-                        if (shouldUseSourcePath)
-                        {
-                            String fullPath = Path.GetFullPath(filename);
-                            resXReader.BasePath = Path.GetDirectoryName(fullPath);
+                            if (shouldUseSourcePath)
+                            {
+                                String fullPath = Path.GetFullPath(filename);
+                                resXReader.BasePath = Path.GetDirectoryName(fullPath);
+                            }
+                            // ReadResources closes the reader for us
+                            ReadResources(reader, resXReader, filename);
                         }
-                        // ReadResources closes the reader for us
-                        ReadResources(reader, resXReader, filename);
-                        break;
-#else
-
-                        using (var xmlReader = new XmlTextReader(filename))
+                        else
+#endif
                         {
-                            xmlReader.WhitespaceHandling = WhitespaceHandling.None;
-                            XDocument doc = XDocument.Load(xmlReader, LoadOptions.PreserveWhitespace);
-                            foreach (XElement dataElem in doc.Element("root").Elements("data"))
+                            if (Traits.Instance.EscapeHatches.UseMinimalResxParsingInCoreScenarios)
+                            {
+                                using (var xmlReader = new XmlTextReader(filename))
+                                {
+                                    xmlReader.WhitespaceHandling = WhitespaceHandling.None;
+                                    XDocument doc = XDocument.Load(xmlReader, LoadOptions.PreserveWhitespace);
+                                    foreach (XElement dataElem in doc.Element("root").Elements("data"))
+                                    {
+                                        string name = dataElem.Attribute("name").Value;
+                                        string value = dataElem.Element("value").Value;
+                                        AddResource(reader, name, value, filename);
+                                    }
+                                }
+                            }
+                            else
                             {
-                                string name = dataElem.Attribute("name").Value;
-                                string value = dataElem.Element("value").Value;
-                                AddResource(reader, name, value, filename);
+                                foreach (IResource resource in MSBuildResXReader.GetResourcesFromFile(filename, shouldUseSourcePath))
```

Comment:
(I assume this is covered by TODO: " Check to ensure that the reader type is available when not targeting netcoreapp3.0", but I want to make sure....)

When we get in here using .NET Core msbuild and we don't have S.R.Extensions (commonly because we're targeting .NETFramework, .NETStandard, or .NET Core 3.0, we need to make sure we support strings (and hopefully resx-file-ref-to-string if possible since that is commonly reported), but reject other types saying S.R.Extensions is needed. Is that the plan?

Reply:
Yes; I'm working on implementing this now. It's going a bit slower than I thought because a bunch of stuff was disabled in the Core build that I need to turn on so I can do the assembly-ref check.

Fileref-to-string should fall out for free, but I'll check it.